transcriptic/analysis/spectrophotometry.py

```diff
@@ -1,132 +1,192 @@
-            if group_well_list and sum([len(i) for i in group_well_list]) > len(data_dict):
+                raise ValueError("Well %s is not in the dataset" % well)
+        elif all(isinstance(i, list) for i in group_wells):
+            if group_wells and sum([len(i) for i in group_wells]) > len(data_dict):
                 raise ValueError("Sum of group lengths exceeds total no. of wells.")
-            for (idx, well_list) in enumerate(group_well_list):
+            for (idx, well_list) in enumerate(group_wells):
                 try:
-                    df_dict[groups[idx]] = [data_dict[humanize(well,well_count,col_count).lower()][0] for well in well_list]
+                    df_dict[group_labels[idx]] = [data_dict[humanize(well,well_count,col_count).lower()][0] for well in well_list]
                 except:
-                    raise ValueError("Well %s is not in the absorbance dataset" % well)
+                    raise ValueError("Well %s is not in the dataset" % well)
         else:
             raise ValueError("Format Error: Group Well List should be a list of list of wells in robot format")
 
         # To ensure pandas dataframe compatiblity: Check that group len elements are of the same length, pad with NaN otherwise
-        if group_well_list and all(isinstance(i, list) for i in group_well_list):
-            group_len_list = [len(x) for x in group_well_list]
+        if group_wells and all(isinstance(i, list) for i in group_wells):
+            group_len_list = [len(x) for x in group_wells]
             if group_len_list.count(group_len_list[0]) != len(group_len_list):
                 max_len = max(group_len_list)
                 for (idx, group_len) in enumerate(group_len_list):
-                    while len(df_dict[groups[idx]]) < max_len:
-                        df_dict[groups[idx]].append(float("NaN"))
+                    while len(df_dict[group_labels[idx]]) < max_len:
+                        df_dict[group_labels[idx]].append(float("NaN"))
 
-        self.df = pandas.DataFrame(df_dict, columns=groups)
-        #If control absorbance object specified, create df_abj variable by subtracting control df from original
+        self.df = pandas.DataFrame(df_dict, columns=group_labels)
 
-        if control_abs:
-            self.df_adj = self.df - control_abs.df
-            # try:
-            #     self.df_adj = self.df - control_abs.df
-            # #Best thing to do here?
-            # except:
-            #     raise ValueError("Control Absorbance object dataframe not compatible with %s dataframe" % self.name)
+        # If control absorbance object specified, create df_abj variable by subtracting control df from original
+        if control_reading:
+            self.df_adj = self.df - control_reading.df
 
         self.cv = self.df.std()/self.df.mean()*100
 
-    def plot(self, mpl=False):
-        # Generates matplotlib obj
+    def plot(self, mpl=False, plot_type="box", **plt_kwargs):
+        """
```

Comment:
pls add a more detailed docstring for plot()... I don't know all the proper lingo :)